swiperefresh/src/main/java/com/google/accompanist/swiperefresh/Slingshot.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.accompanist.swiperefresh
+
+import kotlin.math.abs
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.pow
+
+/**
+ * A utility function that calculates various aspects of 'slingshot' behavior.
+ * Adapted from SwipeRefreshLayout#moveSpinner method.
+ *
+ * @param offsetY The current y offset.
+ * @param maxOffsetY The max y offset.
+ * @param height The height of the item to slingshot.
+ */
+internal fun calculateSlingshot(
```

Comment:
I think we can probably calculate this using a spring, but looks good for now.

Reply:
Yeah that was my thinking and would be ideal to avoid having to maintain this code. I could give it a go in a follow-up PR

Reply:
This isn't critical. Maybe just add a TODO comment so we come back to it